OpenEdXMobile/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -498,6 +502,25 @@ private static TextView createField(@NonNull LayoutInflater inflater, @NonNull V
             put("label", field.getLabel());
             put("value", formattedValue);
         }}));
+
+        // set content description when text is truncated. Truncated text can only be retrieved
+        // the view has been laid out. 
+        textView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
+            @Override
+            public void onGlobalLayout() {
+                if (Build.VERSION.SDK_INT < 16) {
+                    textView.getViewTreeObserver().removeGlobalOnLayoutListener(this);
+                } else {
+                    textView.getViewTreeObserver().removeOnGlobalLayoutListener(this);
+                }
+                String contentDescription = TextUtils.ellipsize(textView.getText(),
+                        textView.getPaint(),
+                        (float) textView.getWidth(),
+                        TextUtils.TruncateAt.END).toString();
+                textView.setContentDescription(contentDescription);
```

Comment:
Why do we want to do this? If truncated text is to be also truncated in TalkBack, maybe we should extend `TextView` to provide this behaviour generally.